Comparing latest changes with master

Randium opened the pull request:
Look at that! Don't merge it, but consider all the changes made so far! Isn't it pweety?

TheretVM committed:
Merge remote-tracking branch 'origin/master' into mine/master

Randium committed:
Merge branch 'Devs/master' into Randium/master

Randium committed:
Merge branch 'Devs/master' into Randium/Flask

Randium committed:
Merge branch 'Devs/master' into Randium/theret

Randium committed:
Show rarity upon refreshing the browser

TheretVM committed:
Fix bugs

Randium committed:
Give less lootboxes to spammers

Randium committed:
Make Vyryn happy

Randium committed:
Merge branch 'Theret' into Randium/Flask

Randium committed:
Merge branch 'Randium/Flask' into Theret

Randium committed:
Merge remote-tracking branch 'Theret/master' into Theret

Randium committed:
Add Russian roulette to Ghost Bot

Randium committed:
Add time command to Time Bot

Randium committed:
Merge branch 'Theret' into Randium/Flask

Randium committed:
Merge branch 'Randium/Flask' into Theret

TheretVM committed:
Fix bugs

Randium committed:
Change User to Ruser

Randium committed:
Add user roulette profile to Ghost Bot

Randium committed:
Check the RIGHT class for None

Randium committed:
Fix layout; add ping

Randium committed:
Reduce timeout to 10 minutes

Randium committed:
Merge branch 'Randium/master' into Theret

Randium committed:
Merge branch 'Theret' into Randium/master

Randium committed:
Merge branch 'Randium/theret' into Theret

Randium committed:
Merge branch 'Theret' into Randium/theret

Randium committed:
Add timeout function

Randium committed:
Merge branch 'Randium/theret' into Theret

TheretVM committed:
Add timeout

TheretVM committed:
Merge remote-tracking branch 'TheretVM/master' into mine/master

TheretVM committed:
Fix bugs

Randium committed:
Add White Werewolf function to fix

Randium committed:
Merge branch 'Theret' into Randium/theret

Randium committed:
Merge branch 'Theret' into Randium/theret

Randium committed:
Merge branch 'Randium/theret' into Theret

Randium committed:
Update roulette scoring system

Randium committed:
Merge branch 'Randium/theret' into Theret

Randium committed:
Give timeout message in correct folder

Randium committed:
Merge branch 'Theret' into Randium/theret

Randium committed:
Insert foreign keys into SQLite database

'cause that's good practice. :3

Randium committed:
Add Devil Bot to framework

Randium committed:
Remove some weird random old database that still went unnoticed

Randium committed:
Rebuild inventory system

Randium committed:
Add rigged pistols to the roulette

Randium committed:
Merge branch 'Devs/devil' into Theret

TheretVM committed:
Fix bugs

TheretVM committed:
Merge remote-tracking branch 'TheretVM/master' into mine/master

Randium committed:
Add lootboxes to inventory (hopefully)

Randium committed:
Merge branch 'Devs/devil' into Theret

Randium committed:
Merge remote-tracking branch 'Theret/master' into Theret

Randium committed:
Add mention to success webhook

This is not very necessary, but it helps a lot in figuring out who used which token at what time.

Randium committed:
Add spy channel integration

From now on, flute players can spy on flute victims! Heheheh.

TheretVM committed:
Fix bugs

TheretVM committed:
Merge remote-tracking branch 'TheretVM/master' into mine/master

TheretVM committed:
Fix bugs

Randium committed:
Ignore list of prizes

Ssssh! It's a secret!

Randium committed:
Allow admins to send custom lootbox gifts

Randium committed:
Reconstruct time shifts

Discord ratelimits should no longer get polls delayed thus far that they will be executed 12 hours later.

Randium committed:
Game Masters can now shift time manually

Randium committed:
Add smite command

Randium committed:
Complete time system + fix bugs

Randium committed:
Merge remote-tracking branch 'Theret/master' into Theret

Randium committed:
Merge branch 'Theret' into Randium/master

Randium committed:
Add refer system

Randium committed:
Welcome new players

Randium committed:
Add more items to lootbox

Randium committed:
Merge branch 'Randium/master' into Theret

Randium committed:
Insert new chest textures

Randium committed:
Allow white werewolf to use actions

Randium committed:
Improve storytime

Randium committed:
Merge branch 'Theret' into Randium/master

Randium committed:
Merge branch 'Randium/master' into Theret

TheretVM committed:
Fix bugs

TheretVM committed:
Add fancy activity visualisation function, including example

TheretVM committed:
Fix bugs

Randium committed:
Add lots of storytime

Randium committed:
Delete file

Randium committed:
JUST GET RID OF THIS SHIT AAAA

Randium committed:
.

Randium committed:
GITKRAKEN SUCKS

Randium committed:
Merge remote-tracking branch 'Theret/master' into Theret

Randium committed:
Merge branch 'Randium/master' into Theret

TheretVM committed:
Fix bugs

Randium committed:
Stop Grandma effect from disappearing before the polls are evaluated

Randium committed:
Merge remote-tracking branch 'Theret/master' into Theret

Randium committed:
Add refer score to profile

TheretVM committed:
Fix bugs

Randium committed:
Fix publish channels bug

Randium committed:
Send poll results to Graveyard

Randium committed:
Fixing grammer error

Randium committed:
Merge remote-tracking branch 'Theret/master' into Theret

TheretVM committed:
Fix bugs

Yes.

Randium committed:
Moving current management to new (old) foolder

Ideally, these will all be cleaned up until only a few files are left.

Randium committed:
Add old folder to PYTHONPATH

Randium committed:
Add easy template for role creation

Randium committed:
Create easy class to quickly view and alter data.

This is gonna make things so much easier.

Randium committed:
Merge branch 'Theret' into Randium/theret

Randium committed:
Build Devil Bot market structure

For now, only the disguise option has been implemented. We're slowly adding things to make sure they all work.

TheretVM committed:
Merge remote-tracking branch 'Randium/theret' into mine/master

TheretVM committed:
Fix bugs

Randium committed:
Merge branch 'Theret' into Randium/theret

Randium committed:
Add invisibility cloak to Devil Bot

Randium committed:
Merge branch 'Theret' into Randium/theret

Randium committed:
Merge branch 'Randium/theret' into Theret

Randium committed:
Add command to view users' inventory for Game Masters

Randium committed:
Merge branch 'Randium/theret' into Theret

Randium committed:
Add Invisibility Cloak to Devil Bot

Randium committed:
Add role, fakerole, demonized, enchanted and more properties

Randium committed:
Fix database bug

Randium committed:
Add unit tests

TheretVM committed:
Fix bugs

Randium committed:
Merge branch 'Theret' into Randium/theret

Randium committed:
Merge branch 'Theret' into Randium/theret

Randium committed:
Update shop table structure

Randium committed:
Change function name to avoid confusion

Randium committed:
Rebuild shop

Randium committed:
Finish shop mechanics

Randium committed:
Fix bugs

Randium committed:
Merge branch 'Randium/theret' into Theret

Randium committed:
Add shop command

TheretVM committed:
Fix bugs

Randium committed:
Merge branch 'Randium/theret' into Theret

Randium committed:
Add shop updates to Time Bot

Randium committed:
Add buy + sell command to Devil Bot

Randium committed:
Fix bugs

Randium committed:
/theret: Auto stash before merge of "Devs/theret" and "Randium/theret"

Randium committed:
Add file to manipulate general database

Randium committed:
Finalize Participant class + add Player class

Randium committed:
Add kill exception wrappers

Randium committed:
Add new roles in new format

Commit:
Create functions to automatically create new commands

Commit:
Add lots of roles to framework

Commit:
Add wrapper to require uses for custom role commands

Commit:
Add role-related functions to player cl;ass

Important to consider is to make these functions for the regular player class as well. This could prevent the function from causing an error when trying to create a non-existing function.

Commit:
Merge branch 'Theret' into Randium/master

Commit:
Add Enums to config file

Commit:
Add requirement wrappers for role-specific commands

Commit:
Add team Enum to roles

Commit:
Add new unittests to repository

Commit:
Add storytime reader for easy storytime manipulation

Commit:
Adding examples/tests to storytime folder

Commit:
Add cc creation channels

Commit:
Add barber attack stories

Commit:
Remove old barber file

Commit:
Merge branch 'Rebuild/dev' into Randium/master

Commit:
Update "deleted" management files

Commit:
Merge branch 'Randium/dev' into Rebuild/dev

Commit:
Allow random storytime messages

Commit:
Merge branch 'Randium/master' into Devs/dev

Commit:
Merge remote-tracking branch 'origin/dev' into Devs/dev

lineargraph commented:
Do i have to say more than this: ![2018-10-21_13-00](https://user-images.githubusercontent.com/20768569/47266052-bf4d8b00-d520-11e8-8460-c5f53229124c.png)

910 spelling warnings and that in a rework made by the grammar nazi himaself.

Randium commented:
Can you gimme some examples?

Commit:
Add cc creation text to storytime

Commit:
Delete outdated test file

Commit:
Attempt to recognize unittests

Commit:
Rebuild Mailbox system

Commit:
Add file for unit testing purposes

Commit:
Add money system to Player class

Commit:
TODO: Add inventory system to Player class

How to claim;
1. Create a new branch;
2. Change the unittest function name to `test_inventory`;
3. Remove the lootbox comment;
4. Write the functions in `management/player.py` in a way that the unittest works. Make sure only to manipulate existing functions with the delivered packages.
4. **(Hint: take a look at the market file that is getting imported)**
5. Make a pull request to this branch.

And woop woop! You're done!

Commit:
TODO: Add unknown players to database

How to claim;
1. Create a new branch;
2. Change the unittest function name to `test_auto_registration`;
3. Remove the lootbox comment;
4. Write the __init__ function in `management/player.py` in a way that the unittest works;
4. Make a pull request to this branch.

And woop woop! You're done!

Commit:
Add time phase stories to story_time folder

Commit:
Finish Grandma role (for now)

OF course, there's stuff to do for that role yet, but the bone structure's there for now.

Commit:
Only allow participants through `require_users` wrapper

Commit:
Add Participant boolean to Player/Participant classes

Commit:
Add poll stories

Commit:
Remove abandoned storytime files

Commit:
Merge branch 'Devs/refactor' into Randium/dev
